core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1323,24 +1314,52 @@ private void commitIndexWriter(IndexWriter writer, Translog translog, String syn
         ensureCanFlush();
         try {
             Translog.TranslogGeneration translogGeneration = translog.getGeneration();
-            final Map<String, String> commitData = new HashMap<>(5);
 
-            commitData.put(Translog.TRANSLOG_GENERATION_KEY, Long.toString(translogGeneration.translogFileGeneration));
-            commitData.put(Translog.TRANSLOG_UUID_KEY, translogGeneration.translogUUID);
+            final String translogFileGen = Long.toString(translogGeneration.translogFileGeneration);
+            final String translogUUID = translogGeneration.translogUUID;
+            final String localCheckpoint = Long.toString(seqNoService().getLocalCheckpoint());
+            final String globalCheckpoint = Long.toString(seqNoService().getGlobalCheckpoint());
 
-            commitData.put(LOCAL_CHECKPOINT_KEY, Long.toString(seqNoService().getLocalCheckpoint()));
-            commitData.put(GLOBAL_CHECKPOINT_KEY, Long.toString(seqNoService().getGlobalCheckpoint()));
+            writer.setLiveCommitData(new Iterable<Map.Entry<String, String>>() {
+                // save the max seq no the first time its computed, so subsequent iterations don't recompute,
+                // potentially getting a different value
+                private String computedMaxSeqNoEntry = null;
```

Comment:
I think it's OK to not use `IndexWriter#getLiveCommitData` - as it may not return what's in the last commit. I suspect this is why it was renamed to say `live` in the name. Is there anything specific you are concerned about?